### ISSUE TITLE
Writing a `Raster` to tif crashes Julia on apple M2

### ISSUE THREAD

**Issue by luraess** [rafaqz/Rasters.jl#343]
I may do something not optimal here, but writing any Raster to disk results in the following `Bus error` leading to the crash of Julia. 

```julia
julia> using Rasters

julia> A = Raster("tos_O1_2001-2002.nc")

julia> B = A[20:30,40:50,1]

julia> write("outB.tif",B)

signal (10): Bus error: 10
in expression starting at REPL[4]:1
unknown function (ip: 0x126e48af8)
Allocations: 42057194 (Pool: 42041755; Big: 15439); GC: 34
```

This is on M2 chip Mac:
```julia
julia> versioninfo()
Julia Version 1.8.2
Commit 36034abf260 (2022-09-29 15:21 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.3.0)
  CPU: 8 × Apple M2
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 4 on 4 virtual cores
Environment:
  JULIA_NUM_THREADS = 4
```

**Comment by rafaqz** [comment]
I can't reproduce this, could it be an apple M2/rosetta problem?

**Comment by luraess** [comment]
Yes. It is certainly a M2 related issue. Where should the issue then be reported precisely?

**Comment by rafaqz** [comment]
Maybe try writing the same thing with NCDatasets.jl without using Rasters.jl. If it happens there too then thats the place.

**Comment by luraess** [comment]
Thanks. Will do so!

**Comment by luraess** [comment]
Interestingly, using only NCDatasets.jl generates an error if not using absolute path
```julia
julia> ds = NCDataset("./tmp/test.nc","c")
ERROR: NetCDF error: Permission denied (NetCDF error code: 13)
Stacktrace:
 [1] check
   @ ~/.julia/packages/NCDatasets/h1epE/src/errorhandling.jl:25 [inlined]
 [2] nc_create(path::String, cmode::UInt16)
   @ NCDatasets ~/.julia/packages/NCDatasets/h1epE/src/netcdf_c.jl:255
 [3] NCDataset(filename::String, mode::String; format::Symbol, share::Bool, diskless::Bool, persist::Bool, memory::Nothing, attrib::Vector{Any})
   @ NCDatasets ~/.julia/packages/NCDatasets/h1epE/src/dataset.jl:222
 [4] NCDataset(filename::String, mode::String)
   @ NCDatasets ~/.julia/packages/NCDatasets/h1epE/src/dataset.jl:163
 [5] top-level scope
   @ REPL[6]:3
```
However, using absolute path works.

Trying to use absolute path in `write` function having loaded `Rasters.jl` still results in 
```julia
julia> write(joinpath(@__DIR__, "out_1.tif"), a)

signal (10): Bus error: 10
in expression starting at REPL[7]:1
unknown function (ip: 0x126e48af8)
Allocations: 52762119 (Pool: 52741466; Big: 20653); GC: 29
zsh: bus error
```

**Comment by luraess** [comment]
Does Rasters.jl directly use `NCDataset ` as `write` function? Since it seems to partly work with `NCDataset`, do you have any idea what could wrong and generate the crash?

**Comment by rafaqz** [comment]
Yes it's using NCDatasets.jl to write. But maybe some different options are triggering this bug. I'll have a look to see what it could be,  but it will be hard without a mac handy.

Is that the full error message? If I can see a stack trace we can mayne find out a bit more.

**Comment by luraess** [comment]
> Is that the full error message? If I can see a stack trace we can maybe find out a bit more.

In Julia yes. I don't know if I can get more out of Bus error from e.g. system logs or so.

**Comment by evetion** [comment]
@luraess Could you test on say, 1.8.1? This feels like https://github.com/JuliaLang/julia/issues/47193, although that turned out to be a problem in ArchGDAL (which was triggered by some changes in Julia itself).

**Comment by evetion** [comment]
In that case, it should be fixed when using ArchGDAL >=0.9.4
